### PR TITLE
UPD: Added basic support for timeout to cli_proton_python.sender

### PR DESCRIPTION
**Pull request by fgiorgetti**

### CODE REVIEW

**Comment by fgiorgetti** [comment]
@pematous I am submitting this PR as I am observing sender instances not stopping when no consumer is connected at the router, so basically despite of using --timeout, sender runs indefinitely. Please take a look if it makes sense.

**Comment by rkubis** [comment]
Hmm, interesting, we have timeout in receiver, but not in sender. It should be there.

File: cli_proton_python/sender.py [hunk 1]
```diff
@@ -33,6 +33,33 @@
...
+    """ Scheduler object for timeout control """
+
+    def __init__(self, parent, event):
+        """
+        Timeout constructor
+
+        saving parent_ptr to get current data
+
+        :param parent: related sender object
+        :param parent: cli_proton_python.Send
```
Comment (jdanekrh) [inline cli_proton_python/sender.py]: should be :type

Comment (fgiorgetti) [reply cli_proton_python/sender.py]: True.

**Comment by fgiorgetti** [comment]
> Hmm, interesting, we have timeout in receiver, but not in sender. It should be there.

@rkubis @jdanekrh One question I have here is. Should the timeout be scheduled only once, when sender is created or should we cancel/re-create every time a message is sent? Basically should we consider timeout as total time to perform the requested operation, or is it message based only?

**Comment by rkubis** [comment]
@fgiorgetti timeout in our QE clients is applied for message. When message is sent/received, timeout should be reset. I think that this behaviour can be seen in receiver.

**Comment by fgiorgetti** [comment]
@rkubis thank you. I will update the PR to cancel/re-create the timeout handler everytime a message gets sent.

**Comment by fgiorgetti** [comment]
@pematous Not sure about moving to CoreClient. The existing logic for handling the timeout is a bit different for sender and receiver. If we decide to have them on CoreClient, then we might go with one of the two approaches: 1 - A parent Timeout (abstract) providing the common properties and leaving on_timer_task to be implemented by concrete classes (TimeoutSend and TimeoutRecv); 2 - Add logic to handle different behavior depending on parent type (Send or Recv). This 2nd approach is simpler, but adds some more complexity. Let me know your thoughts on this.
In the meantime I will prepare some test for it.

**Comment by fgiorgetti** [comment]
@pematous I have a new test ready for this. Once we are done with PR 49, I will rebase it on top of master, as new test depends on the router container added in PR 49 as well as the new ports used in test_conn_urls_send_receive_cli (also from that same PR). So once ready, I will rebase/push again. Thanks.

**Comment by fgiorgetti** [comment]
@pematous I believe it is ok now. Thanks.

**Pull request merged**